metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -155,19 +156,36 @@ def parameter(self, key, value):
         return self
 
 
-class limit(object):
-    def __init__(self, delta_in_secs):
+class Throttle(object):
+    def __init__(self, delta_in_secs=1, num_tries=10):
         self.delta_in_secs = delta_in_secs
+        self.num_tries = num_tries
         self._now = None
+        self._reset()
+
+    def _reset(self):
+        self._tries_left = self.num_tries
+        self._wait = self.delta_in_secs
 
     def __call__(self, func):
         def wrapped(*args, **kwargs):
             now = time.time()
-            if self._now is None or (now - self._now > self.delta_in_secs):
-                func(*args, **kwargs)
+            if self._now is None or (now - self._now > self._wait):
                 self._now = now
+                try:
+                    func(*args, **kwargs)
+                    self._reset()
+                except TriableException as ex:
+                    self._tries_left -= 1
+                    if self._tries_left == 0:
+                        raise ex.ex
+                    self._wait = (self.delta_in_secs*1.1)**(self.num_tries-self._tries_left) + \
```

Comment:
It is a little weird that the jitter portion (from 0 to 3 by default) is larger than the largest exponential wait (1.1^9 = ~2.36)

Reply:
Changed the multiplier to 1.2.